File: Data Structures & Algorithms/validate-parentheses/submission-1.py
class Solution:
    def isValid(self, s: str) -> bool:
        stack = deque()
        map = {
            ')': '(',
            ']': '[',
            '}': '{'
        };

        for char in s:
            if char in map:
                if len(stack) == 0:
                    return False
                
                top = stack.pop()

                if top != map[char]:
                    return False
            
            else:
                stack.append(char)

        
        return len(stack) == 0
            

File: Data Structures & Algorithms/evaluate-reverse-polish-notation/submission-1.js
class Solution {
    /**
     * @param {string[]} tokens
     * @return {number}
     */
    evalRPN(tokens) {
        let stack = []
        for(let i = 0; i < tokens.length; i++){
            switch(tokens[i]){
                case "+": {
                    let fEle = stack.pop()
                    let sEle = stack.pop()

                    stack.push(sEle + fEle)
                    break;
                }

                case "-": {
                    let fEle = stack.pop()
                    let sEle = stack.pop()

                    stack.push(sEle - fEle)
                    break;
                }

                case "*": {
                    let fEle = stack.pop()
                    let sEle = stack.pop()

                    stack.push(sEle * fEle)
                    break;
                }

                case "/": {
                    let fEle = stack.pop()
                    let sEle = stack.pop()
                    let ans = Math.trunc(sEle / fEle)

                    stack.push(ans)
                    break;
                }

                default:
                    let num = Number(tokens[i])
                    stack.push(num)

                    break;
            }
        }

        return stack[0]
    }
}
